server/msteams/client.go

```diff
@@ -200,25 +200,40 @@ func NewManualClient(tenantID, clientID string, logService *pluginapi.LogService
 		client:     nil,
 	}
 
+	// Create an HTTP client with the middleware
```

Comment:
@jespino, can you help me understand the full set of changes to `NewManualClient`? I'm guessing we're lifting some things around here so we can have access to internals that we need for the mocking, but I'm wondering about the risk that our copy/paste here deviates from what the `msgraphsdk` does in the future.

Would it make sense to isolate these changes to creating a `NewManualTestingClient` or some such so that production is guaranteed to be unaffected?

Reply:
Actually, the NewManualClient was already there and the idea was to be used for creating a client manually (with user intervention) that allows to execute the E2E tests that hits the real MSTeams. Probably makes now sense to expose the mockserver to this, and probably we can move it into the tests. Let me review that, but think we doesn't need that changes here. Anyway, I'll do them.